novawallet/Common/Services/ChainRegistry/ChainRegistry.swift

```diff
@@ -123,6 +144,12 @@ final class ChainRegistry {
                 logger?.error("Unexpected error on handling chains update: \(error)")
             }
         }
+        
+        chainsChangesObservers.forEach { observer in
+            observer.queue.async {
```

Comment:
As I see in the chainsSubscribe closure itself already contains dispatch to the concrete queue